### ISSUE TITLE
UX cues on a fresh Wagtail project mislead new users into creating a new page with no content field

### ISSUE THREAD

**Issue by gasman** [wagtail/wagtail#879]
This has caught a couple of people out now, and I've done the same thing myself once or twice when poking around a new Wagtail project on autopilot... https://groups.google.com/d/msg/wagtail/rDtDveWni7U/OkFL86K-tQUJ
- User follows the setup docs and successfully logs in to the admin. Naturally they want to click around to see what they can do here...
- User clicks the topmost menu item, Explorer, which spawns a submenu saying 'Homepage'. Clicks on that too. At this point they're exploring underneath the homepage, which partly obscures the fact that a) there is a page hierarchy and b) the homepage is an actual page that's been created already - they just see a blank list of pages. Which seems reasonable enough, because it's a newly installed site...
- The most prominent thing on the page is a message saying "No pages have been created. Why not add one?". User clicks on it.
- Since there's only one page type available (HomePage), we skip past the 'select a page type' screen and go directly to creating a new page of type HomePage. It's now doubly unclear that what they're actually doing is creating a homepage underneath the existing homepage.
- At this point the user has not been introduced to the idea that multiple page types exist, or that they need to define content fields themselves. They're now expecting to see a big text box where they can enter their page content. Instead, the only thing on the form is a Title field, followed by a big empty space that makes it look like the form hasn't rendered properly.

It's great that we have a UI that draws people in like this rather than leaving them on the dashboard screen scratching their heads, but at the moment it's leading people up the garden path. :-)

**Comment by davecranwell** [comment]
As I see it, the issue is that we've created them a useless homepage. If the homepage were more complete, or if the initial installation had more example models, or if there weren't any models present at all, users would have a less confusing experience.

Given that nothing is possible in wagtail without defining models yourself, it seems that either:

a) we include some more opinionated templates/models intended to work out the box and make the homepage more useful.

This seems a slippery slope. We could provide more detailed example templates/models - I think someone suggested adding a blog at our last meeting - but could we then find ourselves supporting the example models themselves? Didn't we have issues previously with people incapable of *removing* our sample models in wagtaildemo, so they could get started on something real?

b) we remove every model, even the homepage and then deliberately provide messages to the user inside wagtail to cater for the "no models" event.

An example blog model will never be 100% what the user wants, nor will the homepage, so it seems modification of whatever models we provide is inevitable in almost all cases. Providing something the user invariably has to scrap feels like wasted development. Having reached the wagtail admin for the first time, a suitable "Welcome" screen could provide users all the info they need. They'd have to create a model to get started but there's pretty much no other option but to get your hands dirty anyway. We could even provide a sample `models.py` inside the welcome screen itself. 


Not assuming what the user is there to do seems like the best approach, which I'd say involves shipping with no models whatsoever. But that doesn't proclude being helpful too - by providing example models to install if they want them.

**Comment by gasman** [comment]
A suggestion for dealing with this, in the same vein as the signposting info boxes in #1612:

* In the project template, set the homepage to have `subpage_types = []`. (Getting people into the habit of using subpage_types is a good thing, I think...)
* In the explorer view, if all of the following are true:
    * The current page has no children
    * The current page has `subpage_types = []`
    * Only one creatable (non-`is_abstract`) page type is defined
    then in place of the "why not add a page" message, display a developer-oriented message to say something along the lines of: "Welcome to your new Wagtail site! We've created a homepage to get you started - now it's time to define your own page types. See [documentation link] for instructions."

(It might be possible to drop the second and/or third condition, in theory - I believe that, given the way our navigation logic works, there's never actually any other occasion in normal Wagtail use where you see the "there are no pages here - why not add one" message in the explorer. However, I think it's probably good to have these extra conditions in place, to avoid false positives.)

**Comment by timheap** [comment]
This may look weird for single-page sites, which would likely meet all of your criteria for a just-started site.

**Comment by gasman** [comment]
@timheap Good point! Maybe the criteria should be "only one creatable page type exists, and it has no non-core fields defined on it".
